### PR TITLE
Blend the Windows/Unix implementations of ImageAnimator

### LINKED ISSUE #39405: ImageAnimator.StopAnimate (on Unix) could throw PNSE
With dotnet/designs#139 and #39269, we are marking `Thread.Abort()` as Obsolete because it unconditionally throws PlatformNotSupportedException. The tests for the PR turned up that `ImageAnimator.Unix.cs` uses `Thread.Abort()` inside `StopAnimate`.

```csharp
        public static void StopAnimate(Image image, EventHandler onFrameChangedHandler)
        {
            if (image == null)
                return;

            if (ht.ContainsKey(image))
            {
                AnimateEventArgs evtArgs = (AnimateEventArgs)ht[image]!;
                evtArgs.RunThread!.Abort();
                ht.Remove(image);
            }
        }
```

I am going to put a `#pragma warning disable` around the `Thread.Abort()` call, but this code would be unexpectedly throwing PNSE right now, which means it's either dead code or there's a lurking bug that no one has reported.

### PR DESCRIPTION
**Pull request by jeffhandley**
This draft fixes #39405 by reusing the Windows implementation for `ImageAnimator` on Unix platforms. The existing Unix implementation starts a thread per image and relied on aborting those threads when `StopAnimate` was called--`Thread.Abort()` is now obsolete and throws PNSE. The result is that `StopAnimate` currently throws on Linux. The Windows implementation uses a single animation thread, looping through the animating images to update them.

As far as I could tell, there was only one functional difference that prevented the Windows implementation from being used on Linux as-is, and that was the structure of data coming out of `image.GetPropertyItem(PropertyTagFrameDelay)`. On Windows, the length of the byte array was such that every frame's delay was represented. On Linux, for an image with a constant delay between frames, there were only 4 bytes in the array that needed to be repeatedly used for each frame's delay.

I have a test app that I'm using that is performing animations and capturing frames on defined intervals so that I can compare the rendered image before and after this change on both Linux and Windows, but I'm still trying to get the test to be reliable on that timing. I also noticed that there are no unit tests for ImageAnimator; I'm thinking about how I'll want to unit test it.

I wanted to present this draft to see if there are any concerns about taking this approach for the fix.

### CODE REVIEW

**Comment by jkotas** [comment]
Do we need to add or enable any tests?

Looks good to me otherwise.

**Comment by tarekgh** [comment]
LGTM

**Review by danmoseley — APPROVED** [review]

**Comment by jeffhandley** [comment]
Cool; thanks, folks. I'll create some unit tests for ImageAnimator, verify the animation timing through my test app, and then remove `Draft` from the PR.

File: src/libraries/System.Drawing.Common/src/System/Drawing/ImageAnimator.Unix.cs [hunk 1]
```diff
@@ -1,194 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (safern) [inline src/libraries/System.Drawing.Common/src/System/Drawing/ImageAnimator.Unix.cs:1]: Love this 🎉

**Review by safern — APPROVED** [review]

**Comment by jeffhandley** [comment]
While testing this, I found #52144. It turns out, ImageAnimator has really naïve logic for progressing the animation and it's incorrect in several ways. I'm going to close this PR and I'll be submitting a new PR that fixes both #39405 and #52144 with a rewrite of the animation progression logic.

**Pull request closed without merging**